### PR TITLE
Docs: move sphinx-lint to pre-commit

### PR DESCRIPTION
**Pull request by hugovk**
Move sphinx-lint to pre-commit, and so out of the Docs' various requirements.txt files.

Change `make -C Doc check` to run pre-commit instead of only sphinx-lint.

pre-commit is run on the CI in lint.yml, so we no longer need to run `make check` there.

----
:books: Documentation preview :books:: https://cpython-previews--105750.org.readthedocs.build/

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]
I'm not a great reviewer for the change to the Makefile, but everything else LGTM (and I approve of the idea)!

**Comment by hugovk** [comment]
Thanks!

I should also mention I based the Makefile stuff from the peps repo:

https://github.com/python/peps/blob/1b0666eafa3d6ca2c5804f7ed41e49ebb74ca831/Makefile#L62-L66

(Which may have come from similar things in https://github.com/python-pillow/Pillow/blob/main/Makefile)

File: Doc/Makefile [hunk 1]
```diff
@@ -216,11 +216,9 @@ dist:
 	rm dist/python-$(DISTVERSION)-docs-texinfo.tar
 
 .PHONY: check
-check:
-	# Check the docs and NEWS files with sphinx-lint.
-	# Ignore the tools and venv dirs and check that the default role is not used.
-	$(SPHINXLINT) -i tools -i $(VENVDIR) --enable default-role
-	$(SPHINXLINT) --enable default-role ../Misc/NEWS.d/next/
+check: venv
+	$(VENVDIR)/bin/python3 -m pre_commit --version > /dev/null || $(VENVDIR)/bin/python3 -m pip install pre-commit
```
Comment (ezio-melotti) [inline Doc/Makefile:220]: Any reason why this is installed like this instead of adding it to `requirements.txt`?
Everything else LGTM.

Comment (hugovk) [reply Doc/Makefile:220]: Good question, I think to prefer running pre-commit when committing to Git, and avoid installing it unless we really need to - usually we don't need it installed with all the other venv things.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @hugovk

**Comment by CAM-Gerlach** [comment]
Since we also moved the 3.12 branch to using pre-commit and this is just a background infra change, I marked this for backport there too.

**Pull request merged**